servicetalk-http-api/src/main/java/io/servicetalk/http/api/BlockingUtils.java

```diff
@@ -39,54 +34,14 @@ private BlockingUtils() {
         void run() throws Exception;
     }
 
-    interface SupplierCheckedException<T> {
-        T get() throws Exception;
-    }
-
     static Completable blockingToCompletable(RunnableCheckedException r) {
-        return new SubscribableCompletable() {
-            @Override
-            protected void handleSubscribe(final CompletableSource.Subscriber subscriber) {
-                ThreadInterruptingCancellable cancellable = new ThreadInterruptingCancellable(currentThread());
-                subscriber.onSubscribe(cancellable);
-                try {
-                    r.run();
-                } catch (Throwable cause) {
-                    cancellable.setDone(cause);
-                    subscriber.onError(cause);
-                    return;
-                }
-                // It is safe to set this outside the scope of the try/catch above because we don't do any blocking
-                // operations which may be interrupted between the completion of the blockingHttpService call and
-                // here.
-                cancellable.setDone();
-                subscriber.onComplete();
-            }
-        };
-    }
-
-    static <T> Single<T> blockingToSingle(SupplierCheckedException<T> supplier) {
-        return new SubscribableSingle<T>() {
-            @Override
-            protected void handleSubscribe(final SingleSource.Subscriber<? super T> subscriber) {
-                ThreadInterruptingCancellable cancellable = new ThreadInterruptingCancellable(currentThread());
-                subscriber.onSubscribe(cancellable);
-                final T response;
-                try {
-                    response = supplier.get();
-                } catch (Throwable cause) {
-                    cancellable.setDone(cause);
-                    subscriber.onError(cause);
-                    return;
-                }
-                // It is safe to set this outside the scope of the try/catch above because we don't do any blocking
-                // operations which may be interrupted between the completion of the blockingHttpService call and here.
-                cancellable.setDone();
-
-                // The from(..) operator will take care of propagating cancel.
-                subscriber.onSuccess(response);
+        return fromRunnable(() -> {
```

Comment:
An alternative to wrapping is expose some sort of `CheckedRunnable` in the public API, but to just wrap for now.

Reply:
Consider:

```java
    interface RunnableCheckedException {
        void run() throws Exception;

        default void safeRun() {
            try {
                run();
            } catch (Exception e) {
                throwException(e);
            }
        }
    }
```
Then you can do `return fromRunnable(r::safeRun);`